utils/results/results.go

```diff
@@ -0,0 +1,315 @@
+		}
+	}
+	return technologiesSet.ToSlice()
+}
+
+func (sr *TargetResults) GetJasScansResults(scanType jasutils.JasScanType) (results []*sarif.Run) {
+	if sr.JasResults == nil {
+		return
+	}
+	return sr.JasResults.GetResults(scanType)
+}
+
+func (sr *TargetResults) HasInformation() bool {
+	if sr.JasResults != nil && sr.JasResults.HasInformation() {
+		return true
+	}
+	if sr.ScaResults != nil && sr.ScaResults.HasInformation() {
+		return true
+	}
+	return false
+}
+
+func (sr *TargetResults) HasFindings() bool {
+	if sr.JasResults != nil && sr.JasResults.HasFindings() {
+		return true
+	}
+	if sr.ScaResults != nil && sr.ScaResults.HasFindings() {
+		return true
+	}
+	return false
+}
+
+func (sr *TargetResults) AddError(err error) {
+	sr.errorsMutex.Lock()
+	sr.Errors = append(sr.Errors, err)
+	sr.errorsMutex.Unlock()
+}
+
+func (sr *TargetResults) SetDescriptors(descriptors ...string) *TargetResults {
+	if sr.ScaResults == nil {
+		sr.ScaResults = &ScaScanResults{}
+	}
+	sr.ScaResults.Descriptors = descriptors
+	return sr
+}
+
+func (sr *TargetResults) NewScaScanResults(responses ...services.ScanResponse) *ScaScanResults {
+	if sr.ScaResults == nil {
+		sr.ScaResults = &ScaScanResults{}
+	}
+	sr.ScaResults.XrayResults = append(sr.ScaResults.XrayResults, responses...)
+	return sr.ScaResults
+}
+
+func (ssr *ScaScanResults) HasInformation() bool {
+	if ssr.HasFindings() {
+		return true
+	}
+	for _, scanResults := range ssr.XrayResults {
+		if len(scanResults.Licenses) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func (ssr *ScaScanResults) HasFindings() bool {
+	for _, scanResults := range ssr.XrayResults {
+		if len(scanResults.Vulnerabilities) > 0 || len(scanResults.Violations) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func (jsr *JasScansResults) GetResults(scanType jasutils.JasScanType) (results []*sarif.Run) {
+	switch scanType {
+	case jasutils.Applicability:
+		results = jsr.ApplicabilityScanResults
+	case jasutils.Secrets:
+		results = jsr.SecretsScanResults
+	case jasutils.IaC:
+		results = jsr.IacScanResults
+	case jasutils.Sast:
+		results = jsr.SastScanResults
+	}
+	return
+}
+
+func (jsr *JasScansResults) HasFindings() bool {
+	for _, scanType := range jasutils.GetJasScanTypes() {
+		if jsr.HasFindingsByType(scanType) {
+			return true
+		}
+	}
+	return false
+}
+
+func (jsr *JasScansResults) HasFindingsByType(scanType jasutils.JasScanType) bool {
+	for _, run := range jsr.GetResults(scanType) {
+		for _, result := range run.Results {
+			if len(result.Locations) > 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (jsr *JasScansResults) HasInformation() bool {
+	for _, scanType := range jasutils.GetJasScanTypes() {
+		if jsr.HasInformationByType(scanType) {
+			return true
+		}
+	}
+	return false
+}
+
+func (jsr *JasScansResults) HasInformationByType(scanType jasutils.JasScanType) bool {
+	for _, run := range jsr.GetResults(scanType) {
+		if len(run.Results) > 0 {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
what about tests for this file? I see most are getters and setters but shouldn't we test NewScanResults for the locking mechanism, GetTechnologies..
If not disregard this comment